lib/src/private/mli_prv_quant.h

```diff
@@ -125,6 +125,25 @@ inline mli_acc32_t __attribute__ ((always_inline)) weights_additive(
         return init_accum;
 }
 
+//The function uses pointers to pointers for weights. 
+//The caller of the function should compensate for the increment
+//done inside this function.
+template <typename acc_T>
+inline acc_T __attribute__ ((always_inline)) weights_additive_v(
+        const MLI_PTR(int8_t) __restrict *weights, acc_T *init_accum,
+        const s8asym_quant_specific_params* quant_params,
+        const int width,  const int height, int col_step, int row_step) {
+
+    // returns -(in_zero_point * cumsum(weights)) For S8ASYM 
+    if (quant_params->in_offset != 0) {
+        acc_T tmp_acc = reduce_sum2D_v(weights, -quant_params->in_offset, init_accum, width, height, col_step, row_step);
+        //compensite increment of weights pointer from reduce_sum2D_v function
+        weights -= height * row_step;
```

Comment:
This code has no effect because * is missing before weights. but I'm not sure if the compensation should be here or at the caller of weithgs_additive_v? because that is what is written in the comments.